crates/turborepo-lib/src/diagnostics.rs

```diff
@@ -0,0 +1,492 @@
+                                Self::failed(chan, "Git FS Monitor not configured".to_string())
+                                    .await;
+                                return;
+                            }
+                            Ok(_) => unreachable!(),
+                            Err(_) => {
+                                // the sender (terminal) was shut, ignore
+                            }
+                        }
+                    } else {
+                        Self::log_line(&chan, "Git FS Monitor settings set".to_string()).await;
+                    }
+                }
+                Ok(_) => unreachable!(), // the vec of futures has exactly 3 elements
+                Err(e) => {
+                    Self::failed(chan, format!("Failed to get git version: {}", e)).await;
+                    return;
+                }
+            }
+
+            Self::done(chan, "Git FS Monitor Enabled".to_string()).await;
+        });
+    }
+}
+
+pub struct DaemonDiagnostic(pub AbsoluteSystemPathBuf);
+
+impl Diagnostic for DaemonDiagnostic {
+    const NAME: &'static str = "turbo.daemon";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        let daemon_root = self.0.clone();
+        tokio::task::spawn(async move {
+            Self::started(&chan, "Turbo Daemon".to_string()).await;
+            Self::log_line(&chan, "Connecting to daemon...".to_string()).await;
+            Self::log_line(&chan, "Getting status...".to_string()).await;
+
+            let connector = DaemonConnector {
+                can_kill_server: false,
+                can_start_server: true,
+                pid_file: daemon_root.join_component("turbod.pid"),
+                sock_file: daemon_root.join_component("turbod.sock"),
+            };
+
+            let mut client = match connector.connect().await {
+                Ok(client) => client,
+                Err(e) => {
+                    Self::failed(chan, format!("Failed to connect to daemon: {}", e)).await;
+                    return;
+                }
+            };
+
+            match client.status().await {
+                Ok(status) => {
+                    Self::log_line(&chan, format!("Daemon up for {}ms", status.uptime_msec)).await;
+                    Self::done(chan, "Daemon is running".to_string()).await;
+                }
+                Err(e) => {
+                    Self::failed(chan, format!("Failed to get daemon status: {}", e)).await;
+                }
+            }
+        });
+    }
+}
+
+pub struct LSPDiagnostic(pub AbsoluteSystemPathBuf);
+impl Diagnostic for LSPDiagnostic {
+    const NAME: &'static str = "turbo.lsp";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        let daemon_root = self.0.clone();
+        tokio::task::spawn(async move {
+            Self::started(&chan, "Turborepo Extension".to_string()).await;
+            Self::log_line(&chan, "Checking if it is running...".to_string()).await;
```

Comment:
nit, this is less clear than I'd like
<img width="491" alt="Screenshot 2023-12-13 at 12 03 04 PM" src="https://github.com/vercel/turbo/assets/7357863/d05d840f-904e-4efa-8047-6bbcf0918c17">

```suggestion
            Self::log_line(&chan, "Checking if extension is running...".to_string()).await;
```